src/main/java/org/dice_research/opal/civet/metrics/DescriptionMetricImproved.java

```diff
@@ -0,0 +1,165 @@
+			+ "else if dct:description contains >20 nouns, verbs or adjectives then 5 stars"
+			+ "else if no dct:description but dct:title then evaluate title and give stars";
+
+	// German parts of speech tags for noun, verb and adjective
+	private final String nounTag = "NN";
+	private final String verbTag = "VVFIN";
+	private final String adjectiveTag = "ADJA";
+	int countNouns = 0;
+	int lastIndexNoun = 0;
+	int countVerbs = 0;
+	int lastIndexVerb = 0;
+	int countAdjectives = 0;
+	int lastIndexAdjective = 0;
+
+	public int posTagger(String rdfPredicate) throws IOException {
+
+		// Loading Parts of speech-maxent model
+		InputStream inputStream = new FileInputStream("src/main/resources/de-pos-maxent.bin");
+		POSModel model = new POSModel(inputStream);
+
+		// Instantiating POSTaggerME class
+		POSTaggerME tagger = new POSTaggerME(model);
+
+		// Tokenizing the sentence using WhitespaceTokenizer class
+		WhitespaceTokenizer whitespaceTokenizer = WhitespaceTokenizer.INSTANCE;
+		String[] tokens = whitespaceTokenizer.tokenize(rdfPredicate);
+
+		// Generating tags
+		String[] tags = tagger.tag(tokens);
+
+		// Instantiating the POSSample class
+		POSSample predicateTags = new POSSample(tokens, tags);
+		String posTags = predicateTags.toString();
+
+		// Counting occurrences of nouns
+		while (lastIndexNoun != -1) {
+			lastIndexNoun = posTags.indexOf(nounTag, lastIndexNoun);
+
+			if (lastIndexNoun != -1) {
+				countNouns++;
+				lastIndexNoun += nounTag.length();
+			}
+		}
+
+		// Counting occurrences of verbs
+		while (lastIndexVerb != -1) {
+			lastIndexVerb = posTags.indexOf(verbTag, lastIndexVerb);
+
+			if (lastIndexVerb != -1) {
+				countVerbs++;
+				lastIndexVerb += verbTag.length();
+			}
+		}
+
+		// Counting occurrences of adjectives
+		while (lastIndexAdjective != -1) {
+			lastIndexAdjective = posTags.indexOf(adjectiveTag, lastIndexAdjective);
+
+			if (lastIndexAdjective != -1) {
+				countAdjectives++;
+				lastIndexAdjective += adjectiveTag.length();
+			}
+		}
+
+		// Sum up all posTags in one variable to rate the metric
+		int allPosTags = countNouns + countVerbs + countAdjectives;
+
+		// Now rating for the description is given
+		if (allPosTags <= 5) {
+			// Very less posTags
+			return 1;
+		}
+
+		else if (allPosTags <= 10) {
+			// less posTags
+			return 2;
+		}
+
+		else if (allPosTags <= 15) {
+			// average posTags
+			return 3;
+		}
+
+		else if (allPosTags <= 20) {
+			// above average posTags
+			return 4;
+		}
+
+		else {
+			// Best use of posTags
+			return 5;
+		}
+	}
+
+	@Override
+	public Integer compute(Model model, String datasetUri) throws Exception {
+		int score = 1;
+		Resource dataset = model.createResource(datasetUri);
+		if (!(dataset.hasProperty(DCTerms.description)) && (dataset.hasProperty(DCTerms.title))) {
+			String title = dataset.getProperty(DCTerms.title).getObject().toString();
+			score = posTagger(title);
+			return score;
```

Comment:
no need to type return sure, as you are returning score after if-else any way

Reply:
Done.